corehq/apps/app_manager/app_translations/generators.py

```diff
@@ -92,7 +96,63 @@ def _set_sheet_name_to_module_or_form_mapping(self, all_module_and_form_details)
                 row[unique_id_column_index]
             )
 
+    def _generate_module_sheet_name(self, module_index):
+        """
+        receive index of module and convert into name with module unique id
+
+        :param module_index: index of module in the app
+        :return: name like module_moduleUniqueId
+        """
+        _module = self.app.modules[module_index]
+        module_unique_id = _module.unique_id
+        sheet_name = "_".join(["module", module_unique_id])
+        self.slug_to_name[sheet_name] = _module.name
+        return sheet_name
+
+    def _generate_form_sheet_name(self, module_index, form_index):
+        """
+        receive index of form and module and convert into name with form unique id
+
+        :param module_index: index of form's module in the app
+        :param form_index: index of form in the module
+        :return: name like form_formUniqueId
+        """
+        form = self.app.modules[module_index].forms[form_index]
+        form_unique_id = form.unique_id
+        sheet_name = "_".join(["form", form_unique_id])
+        self.slug_to_name[sheet_name] = form.name
+        return sheet_name
+
+    def _update_sheet_name_with_unique_id(self, sheet_name):
+        """
+        update sheet name with HQ format like module0 or module1_form1 to
+        a name with unique id of module or form instead
+
+        :param sheet_name: name like module0 or module1_form1
+        :return: name like module_moduleUniqueID or form_formUniqueId
+        """
+        if sheet_name == MODULES_AND_FORMS_SHEET_NAME:
+            return sheet_name
+        module_sheet_name_match = HQ_MODULE_SHEET_NAME.match(sheet_name)
+        if module_sheet_name_match:
+            module_index = int(module_sheet_name_match.groups()[0]) - 1
+            return self._generate_module_sheet_name(module_index)
+        form_sheet_name_match = HQ_FORM_SHEET_NAME.match(sheet_name)
+        if form_sheet_name_match:
+            indexes = form_sheet_name_match.groups()
+            module_index, form_index = int(indexes[0]) - 1, int(indexes[1]) - 1
+            return self._generate_form_sheet_name(module_index, form_index)
+        raise Exception("Got unexpected sheet name %s" % sheet_name)
+
     def _get_filename(self, sheet_name):
+        """
+        receive sheet name in HQ format and return the name that should be used
+        to upload on transifex along with module/form unique ID and version postfix
```

Comment:
> Any idea how hard it'd be to switch that to use unique IDs too

Use unique IDs in HQ?
I think It makes more sense for user to see "Module0_form1" as the sheet name instead of ids in the excel sheet.

> to make the intermediate format contain both IDs? 

You mean hq to give both ids already? i am not sure how useful ID would be anywhere else so i would avoid adding that to HQ.